src/Connections/GitlabRepo.ts

```diff
@@ -386,12 +386,8 @@ export class GitLabRepoConnection extends CommandConnection<GitLabRepoConnection
         skip?: boolean,
     }>();
 
-    /**
-     * GitLab provides NO threading information in its webhook response objects,
-     * so we need to determine if we've seen a comment for a line before, and
-     * skip it if we have (because it's probably a reply).
-     */
-    private readonly mergeRequestSeenDiscussionIds = new QuickLRU<string, undefined>({ maxSize: 100 });
+    // TODO persistence, somehow
+    private readonly discussionThreads = new QuickLRU<string, Promise<string|undefined>>({ maxSize: 100});
```

Comment:
Pop it in a bridge storage :). Set a LRU on redis. Use QuickLRU for memory store.